openhtf/exe/test_state.py

```diff
@@ -116,17 +116,13 @@ def GetFinishedRecord(self):
     """
     self.logger.debug('Finishing test execution with state %s.', self._state)
 
-    if 'config' in self.record.metadata:
-      self.logger.warning('config already set in metadata, not saving config')
-    else:
-      self.record.metadata['config'] = conf.Config()
-
     if not self.record.dut_id:
       raise BlankDutIdError(
           'Blank or missing DUT ID, HTF requires a non-blank ID.')
 
-    return self.record._replace(
-        end_time_millis=util.TimeMillis(), outcome=self._state)
+    self.record.end_time_millis = util.TimeMillis()
```

Comment:
I like having record populated while the test is running because then it can be repeatedly serialized and served by the frontend API instead of having the frontend API build all the same stuff manually. Why not just have GetFinishedRecord() raise if end_time_millis isn't already set?